morty: init -> 0.2.0

leenaars opened the pull request:
###### Motivation for this change

This tool complements [searx](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/networking/searx.nix) (from the same author), and provides a proxy service that can be used for visiting search results from the search engine. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

grahamc commented:
@GrahamcOfBorg eval

grahamc commented on nixos/tests/morty.nix:
```diff
@@ -0,0 +1,33 @@
+          enable = true;
+	  key = "78a9cd0cfee20c672f78427efb2a2a96036027f0";
+	  port = 3001;
+	};
```
Please unindent `enable` by 2 spaces, then indent the rest to be in line with the enable line.

grahamc commented on pkgs/servers/web-apps/morty/default.nix:
```diff
@@ -0,0 +1,40 @@
+      * No Caching/Etag
+      * Supports GET/POST forms and IFrames
+      * Optional HMAC URL verifier key to prevent service abuse
+   ";
```
Please use `''` instead of `"` (use two single quotes instead of one double-quote) on long text blocks. Also, move the first line of the text to its own line. Finally, the `maintainer` line should be `maintainers`.

joachifm commented on nixos/modules/misc/ids.nix:
```diff
@@ -576,6 +577,7 @@
       mighttpd2 = 285;
       hass = 286;
       monero = 287;
+      morty = 288;
```
Why does this need a static uid?

leenaars replied:
I'm new at packaging services, and basically replicated the setup of searx which is its companion tool. That set uid and gid, so I did that for morty as well.
 
Is there a checklist for adding services somewhere?

joachifm replied:
Not really; I'm just wondering whether there's a reason for allocating a uid; if there's a reason that's totally fine, otherwise you can probably get away with dynamic uids.

leenaars replied:
I'm certainly not an expert in this. If you want this removed, let me know.

joachifm replied:
What type of data does the service generate? Are you likely to transfer it across hosts & is it too expensive to fixup permissions at service start or do you need to refer to the uid before the uid/name mapping is established? If so, using a static uid may be appropriate. I'm happy to defer to your judgement regarding the service, I know nothing about it, but often static uids are used for no good reason and so the pattern propagates.

leenaars commented:
I removed the fixed uid/gid mapping, under the assumption that a proxy has little worthwhile state.

joachifm commented:
@GrahamcOfBorg test morty

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34886&attempt_id=1969fafc-3c70-4487-984a-92b5097a6716)

<details><summary>Partial log (click to expand)</summary><p>

```
while evaluating ‘dischargeProperties’ at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ogden/lib/modules.nix:386:25, called from /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ogden/lib/modules.nix:324:62:
while evaluating the attribute ‘condition’ at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ogden/lib/modules.nix:467:14:
while evaluating the attribute ‘condition’ at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ogden/lib/modules.nix:467:14:
while evaluating the attribute ‘condition’ at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ogden/lib/modules.nix:467:14:
while evaluating the attribute ‘services.nixosManual’ at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ogden/lib/attrsets.nix:199:44:
while evaluating anonymous function at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ogden/lib/modules.nix:75:45, called from /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ogden/lib/attrsets.nix:199:52:
while evaluating ‘yieldConfig’ at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ogden/lib/modules.nix:74:29, called from /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ogden/lib/modules.nix:77:16:
while evaluating anonymous function at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ogden/lib/modules.nix:80:24, called from /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ogden/lib/modules.nix:80:11:
while evaluating anonymous function at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ogden/lib/modules.nix:81:26, called from /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ogden/lib/modules.nix:81:13:
The option `services.morty' defined in `<unknown-file>' does not exist.
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34886&attempt_id=0beb4506-df9a-4b99-a3f7-53bd568aed8e)

<details><summary>Partial log (click to expand)</summary><p>

```
while evaluating 'dischargeProperties' at /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/lib/modules.nix:386:25, called from /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/lib/modules.nix:324:62:
while evaluating the attribute 'condition' at /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/lib/modules.nix:467:14:
while evaluating the attribute 'condition' at /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/lib/modules.nix:467:14:
while evaluating the attribute 'condition' at /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/lib/modules.nix:467:14:
while evaluating the attribute 'services.nixosManual' at /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/lib/attrsets.nix:199:44:
while evaluating anonymous function at /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/lib/modules.nix:75:45, called from /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/lib/attrsets.nix:199:52:
while evaluating 'yieldConfig' at /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/lib/modules.nix:74:29, called from /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/lib/modules.nix:77:16:
while evaluating anonymous function at /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/lib/modules.nix:80:24, called from /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/lib/modules.nix:80:11:
while evaluating anonymous function at /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/lib/modules.nix:81:26, called from /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/lib/modules.nix:81:13:
The option `services.morty' defined in `<unknown-file>' does not exist.
```
</p></details>

Commit:
php: 7.2.5 -> 7.2.6

Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools.

This update was made based on information from https://repology.org/metapackage/php/versions.

These checks were done:

- built on NixOS
- /nix/store/25l2hz7njpg9glpmslcadkgqwai5f77s-php-7.2.6/bin/phar.phar passed the binary check.
- /nix/store/25l2hz7njpg9glpmslcadkgqwai5f77s-php-7.2.6/bin/phar passed the binary check.
- /nix/store/25l2hz7njpg9glpmslcadkgqwai5f77s-php-7.2.6/bin/php passed the binary check.
- /nix/store/25l2hz7njpg9glpmslcadkgqwai5f77s-php-7.2.6/bin/phpdbg passed the binary check.
- /nix/store/25l2hz7njpg9glpmslcadkgqwai5f77s-php-7.2.6/bin/php-cgi passed the binary check.
- /nix/store/25l2hz7njpg9glpmslcadkgqwai5f77s-php-7.2.6/bin/pear passed the binary check.
- /nix/store/25l2hz7njpg9glpmslcadkgqwai5f77s-php-7.2.6/bin/peardev passed the binary check.
- /nix/store/25l2hz7njpg9glpmslcadkgqwai5f77s-php-7.2.6/bin/pecl passed the binary check.
- /nix/store/25l2hz7njpg9glpmslcadkgqwai5f77s-php-7.2.6/bin/php-fpm passed the binary check.
- 9 of 9 passed binary check by having a zero exit code.
- 0 of 9 passed binary check by having the new version present in output.
- found 7.2.6 with grep in /nix/store/25l2hz7njpg9glpmslcadkgqwai5f77s-php-7.2.6
- directory tree listing: https://gist.github.com/409d2cfaa7e805714825281fbaba0d0f
- du listing: https://gist.github.com/7fbd8e3d56524f70b3dfb94c045fccd2

joachifm commented:
@GrahamcOfBorg test morty

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34886&attempt_id=5e9e24da-8328-415e-9722-91a90d62a722)

The following builds were skipped because they don't evaluate on aarch64-linux: tests.morty

<details><summary>Partial log (click to expand)</summary><p>

```
while evaluating the attribute 'values' at [1m/var/lib/gc-of-borg/nix-test-rs-12/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-12/lib/modules.nix[0m:434:7:
while evaluating 'concatMap' at [1m/var/lib/gc-of-borg/nix-test-rs-12/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-12/lib/lists.nix[0m:104:18, called from [1m/var/lib/gc-of-borg/nix-test-rs-12/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-12/lib/modules.nix[0m:434:16:
while evaluating 'concatMap' at [1m/var/lib/gc-of-borg/nix-test-rs-12/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-12/lib/lists.nix[0m:104:18, called from [1m/var/lib/gc-of-borg/nix-test-rs-12/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-12/lib/modules.nix[0m:324:17:
while evaluating anonymous function at [1m/var/lib/gc-of-borg/nix-test-rs-12/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-12/lib/modules.nix[0m:324:28, called from undefined position:
while evaluating 'dischargeProperties' at [1m/var/lib/gc-of-borg/nix-test-rs-12/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-12/lib/modules.nix[0m:392:25, called from [1m/var/lib/gc-of-borg/nix-test-rs-12/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-12/lib/modules.nix[0m:325:62:
while evaluating the attribute 'value' at [1m/var/lib/gc-of-borg/nix-test-rs-12/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-12/lib/modules.nix[0m:216:48:
while evaluating the attribute 'config.uid' at [1m/var/lib/gc-of-borg/nix-test-rs-12/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-12/nixos/modules/services/networking/morty.nix[0m:76:9:
while evaluating the attribute 'ids.uids.morty' at [1m/var/lib/gc-of-borg/nix-test-rs-12/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-12/lib/attrsets.nix[0m:199:44:
attribute 'morty' missing, at [1m/var/lib/gc-of-borg/nix-test-rs-12/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-12/nixos/modules/services/networking/morty.nix[0m:76:15

```
</p></details>

GrahamcOfBorg commented:
No attempt on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34886&attempt_id=fab2ff0e-21e9-4b2e-83aa-5e2411e86906)

The following builds were skipped because they don't evaluate on x86_64-linux: tests.morty

<details><summary>Partial log (click to expand)</summary><p>

```
while evaluating the attribute 'values' at /home/ofborg/ofborg3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-7c6f434c3/lib/modules.nix:434:7:
while evaluating 'concatMap' at /home/ofborg/ofborg3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-7c6f434c3/lib/lists.nix:104:18, called from /home/ofborg/ofborg3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-7c6f434c3/lib/modules.nix:434:16:
while evaluating 'concatMap' at /home/ofborg/ofborg3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-7c6f434c3/lib/lists.nix:104:18, called from /home/ofborg/ofborg3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-7c6f434c3/lib/modules.nix:324:17:
while evaluating anonymous function at /home/ofborg/ofborg3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-7c6f434c3/lib/modules.nix:324:28, called from undefined position:
while evaluating 'dischargeProperties' at /home/ofborg/ofborg3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-7c6f434c3/lib/modules.nix:392:25, called from /home/ofborg/ofborg3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-7c6f434c3/lib/modules.nix:325:62:
while evaluating the attribute 'value' at /home/ofborg/ofborg3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-7c6f434c3/lib/modules.nix:216:48:
while evaluating the attribute 'config.uid' at /home/ofborg/ofborg3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-7c6f434c3/nixos/modules/services/networking/morty.nix:76:9:
while evaluating the attribute 'ids.uids.morty' at /home/ofborg/ofborg3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-7c6f434c3/lib/attrsets.nix:199:44:
attribute 'morty' missing, at /home/ofborg/ofborg3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-7c6f434c3/nixos/modules/services/networking/morty.nix:76:15

```
</p></details>

joachifm commented on nixos/modules/services/networking/morty.nix:
```diff
@@ -0,0 +1,103 @@
+  config = mkIf config.services.morty.enable {
+
+    users.extraUsers.morty =
+      { uid = config.ids.uids.morty;
```
Eval error here & below. Just leave unspecified to have ids dynamically allocated.

leenaars replied:
Fixed this, and also updated to fresh release.

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34886&attempt_id=a34c86fb-0bf6-44ca-87e3-a1b5d92b831d)

Attempted: tests.morty

<details><summary>Partial log (click to expand)</summary><p>

```
mortyProxyWithKey: running command: sync
mortyProxyWithKey# [   17.568988] dhcpcd[765]: eth0: Router Advertisement from fe80::2
mortyProxyWithKey# [   17.577832] dhcpcd[765]: eth0: adding address fec0::5054:ff:fe12:3456/64
mortyProxyWithKey: exit status 0
test script finished in 18.50s
cleaning up
killing mortyProxyWithKey (pid 627)
vde_switch: EOF on stdin, cleaning up and exiting
vde_switch: Could not remove ctl dir '/build/vde1.ctl': Directory not empty
/nix/store/3lchhnjr9zyimaj62j3g9whz0shqbzx7-vm-test-run-morty
```
</p></details>

joachifm commented:
LGTM, though to be strictly conforming to the guidelines this should ideally be one commit for the package and one for the service, with appropriate commit subjects.

leenaars committed:
morty: init -> 0.2.0

leenaars committed:
morty: init as service

leenaars commented:
@joachifm: split into one commit for the package and one for the service.
